src/HexagonOptimize.cpp

```diff
@@ -112,11 +112,17 @@ struct Pattern {
         ExactLog2Op1 = 1 << 3, // Replace operand 1 with its log base 2, if the log base 2 is exact.
         ExactLog2Op2 = 1 << 4, // Save as above, but for operand 2.
 
+        FirstExactLog2Op = 1,   // FirstExactLog2Op and NumExactLog2Op ensure that we check only op1 and op2
```

Comment:
I am wondering if the flags necessary to describe vmpa are worth it. I think it would be overall a smaller amount of code that is easier to understand to just have a custom piece of code in `OptimizePatterns::visit(const Add *)` that just directly did its own pattern matching and checked if the matched values are suitable for vmpa with some hard-coded logic. It also seems unlikely that these extra flags will be used by anything *but* vmpa, so if they are only used once, that also suggests just hard-coding the logic in the Add visitor.